x-pack/plugins/maps/public/layers/vector_layer.js

```diff
@@ -793,9 +744,55 @@ export class VectorLayer extends AbstractLayer {
     }
   }
 
+  _createTextLayerSource() {
+    const featureCollection = this._getSourceFeatureCollection();
+    if (!featureCollection || !featureCollection.hasOwnProperty('features')) return;
+    const largestFeatures = featureCollection.features.map(feature => {
```

Comment:
Alternatively, we could use `[@turf/centerOfMass](http://turfjs.org/docs/#centerOfMass)` or `[polylabel](https://github.com/mapbox/polylabel)` here to create a point source for the textLayer as opposed to cloning what could be very large polygons. However, the position of the label would be static and possibly off-screen when zoomed in.